### PR TITLE
e2e: move funs of framework/job to e2e/upgrades/apps/job

### PR DESCRIPTION
**Pull request by haosdent**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Move funs of framework/job to e2e/upgrades/apps/job.

**Which issue(s) this PR fixes**:
Ref #86052

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: test/e2e/upgrades/apps/job.go [hunk 1]
```diff
@@ -17,8 +17,14 @@ limitations under the License.
...
 import (
+	"fmt"
+	"strings"
+
 	batchv1 "k8s.io/api/batch/v1"
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	clientset "k8s.io/client-go/kubernetes"
```
Comment (oomichi) [inline test/e2e/upgrades/apps/job.go]: The PR job is failed like:
```
Execution platform: //:rbe_with_network
compilepkg: missing strict dependencies:
	/b/f/w/test/e2e/upgrades/apps/job.go: import of "k8s.io/client-go/kubernetes"
No dependencies were provided.
Check that imports in Go sources match importpath attributes in deps.
```
Need to run `./hack/update-bazel.sh` and apply the change into this PR.

Comment (haosdent) [reply test/e2e/upgrades/apps/job.go]: sry, fixed. thanks a lot for @oomichi help! Could you help to review again?

**Comment by oomichi** [comment]
Thanks for updating

/retest
/lgtm

**Pull request merged**